### PR TITLE
feat: adding new schema for pharmacogenomics

### PR DESCRIPTION
**Pull request by DSuveges**
The pydantic schema and the resulting json representation is ready. However I would wait with the review until there's a chance to run against the full dataset.

### CODE REVIEW

File: pydantic_models/pharmacogenomics.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    genotype: str = Field(
+        description="Genotype string", examples=["(CA)16/(CA)17", "TA", "del/GAG"]
+    )
+    genotypeAnnotationText: str = Field(
+        description="Full annotation string for genotype.",
+        examples=[
+            "Patients with the rs121918596 del/GAG genotype may develop malignant hyperthermia when treated with volatile anesthetics [...]"
+        ],
+    )
+    drugFromSource: str = Field(description="Drug name", examples=["succinylcholine"])
```
Comment (tskir) [inline pydantic_models/pharmacogenomics.py:86]:
```suggestion
    drugFromSource: str = Field(description="Drug name.", examples=["succinylcholine"])
```

File: pydantic_models/pharmacogenomics.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    drugId: Optional[str] = Field(
+        description="CHEBI ID of drug, mapped through OLS",
```
Comment (tskir) [inline pydantic_models/pharmacogenomics.py:88]:
```suggestion
        description="CHEBI ID of drug, mapped through OLS.",
```

File: pydantic_models/pharmacogenomics.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+class phenotypeCategory(str, Enum):
+    """Accepted phenotype categories describing pharmacogenomic effect."""
+
+    toxicity = "toxicity"
+    efficacy = "efficacy"
+    dosage = "dosage"
+    metabolism = ("metabolism/pk",)
+    pd = "pd"
+    other = "other"
```
[the remark this answers is not in the document]
Comment (DSuveges) [reply pydantic_models/pharmacogenomics.py:18]: That's correct. It is not clear to me, if we can/should use an enum for `phenotypeCategory`. I wanted to have it as a placeholder and get back to it later. Although we have only a few possible terms, sometimes there are more than one of them. But not convinced we should use a list...

File: pydantic_models/pharmacogenomics.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    )
+    evidenceLevel: EvidenceLevel = Field(
+        description="Level of evidence. More info: https://www.pharmgkb.org/page/clinAnnLevels.",
+        examples=["1A"],
+    )
+    literature: List[str] = Field(
+        description="List of PMIDs of supporting publications.",
+    )
+    genotypeId: Optional[str] = Field(
+        description="VCF-style (chr_pos_ref_allele1,allele2) identifier of genotype; computed as described here: https://github.com/apriltuesday/opentargets-pharmgkb/tree/issue-18#variant-coordinate-computation.",
```
[the remark this answers is not in the document]
Comment (DSuveges) [reply pydantic_models/pharmacogenomics.py:54]: Definitely.

File: pydantic_models/pharmacogenomics.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tskir) [reply pydantic_models/pharmacogenomics.py:18]: Sounds good!

**Review by tskir — APPROVED** [review]
Everything's good now, thank you Daniel!

**Pull request merged**